spec/code-snippets/DeclarationReferences.ts

```diff
@@ -0,0 +1,443 @@
+// for referencing other API declarations such as classes, member functions, enum values,
+// etc.  It is used with standard TSDoc tags such as {@link} and {@inheritDoc}.
+//
+// The full declaration reference syntax supports declarations that are imported from
+// another NPM package, optionally with an NPM scope, optionally with an explicit
+// import path.
+//
+// For example:
+//
+// /**
+//  * {@link @my-scope/my-package/path1/path2:namespace1.namespace2.MyClass.myMember
+//  * | a complex example}
+//  */
+//
+// The optional components to the left of the ":" simply follow the standard rules of
+// a TypeScript "import" definition, so we don't discuss them further here.
+//
+// TSDoc declaration references are always resolved relative to a specific entry point
+// (NOT relative to the current source file or declaration scope).  Thus, their syntax
+// is independent of where the reference occurs within a given package.  For example,
+// when referring to "MyClass.methodA" below, we may want to shorten it to "methodA"
+// (since the meaning seems clear from context), but TSDoc standard does not allow that.
+// "MyClass.methodA" is the shortest allowable name:
+//
+// export class MyClass {
+//   public methodA(): void {
+//   }
+//
+//   /**
+//    * CORRECT:     {@link MyClass.methodA}
+//    * NOT CORRECT: {@link methodA}
+//    */
+//   public methodB(): void {
+//   }
+// }
+//
+// Requiring fully scoped names ensures that documentation processors can resolve links
+// efficiently and without access to a compiler analysis.  (For this document, it means
+// all our examples can be simple self-references, since cross-references would look the same.)
+//
+// The most interesting feature of this syntax is the "[]" selector, which has these properties:
+//
+// - It is used to disambiguate an already traversed node (e.g. choosing a function overload)
+//
+// - It is NEVER used to traverse into a new node (e.g. we write MyClass."member with spaces"
+//   instead of MyClass["member with spaces"]
+//
+// - For classes, the supported selectors are "[instance]", "[static]", or "[constructor]"
+//
+// - For merged declarations, the supported selectors are "[class]", "[enum]", "[function]",
+//   "[interface]", "[namespace]", "[type]", or "[variable]"
+//
+// - When selecting a user-defined label (e.g. "[WITH_NUMBERS]") the label must be
+//   all capitals, which ensures that it cannot conflict with a TSDoc reserved name such
+//   as "class".
+
+
+//---------------------------------------------------------
+// Static vs instance members
+
+/**
+ * Shortest name:  {@link Class00}
+ * Full name:      {@link Class00[class]}
+ */
+export class Class00 {
+  /**
+   * Shortest name:  {@link Class00.member01[instance]}
+   * Full name:      {@link Class00[class].member01[instance]}
+   */
+  public member01(): void {
+  }
+
+  /**
+   * Shortest name:  {@link Class00.member01[static]}
```

Comment:
>static [](start = 45, length = 6)

It'd be useful to have an example where there isn't both a static and instance member with the same name.